junit-plugin/src/main/scala/scala/scalanative/junit/plugin/ScalaNativeJUnitPlugin.scala

```diff
@@ -174,6 +174,14 @@ class ScalaNativeJUnitPlugin(val global: Global) extends NscPlugin {
           .map(gen.mkMethodCall(Ident(module), _, Nil, Nil))
           .toList
 
+        val nonPublicCalls = calls.filterNot(_.symbol.isPublic)
```

Comment:
Doing this after having created all the `calls` means that we're still temporarily generating invalid code. I suggest `partition`ing the `annotatedMethods` according to whether they are public first. Then report errors on the non-public ones, and only generate `mkMethodCall`s for the public ones.